flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/history/FlinkJsonTableChangeSerializer.java

```diff
@@ -0,0 +1,208 @@
+
+        return Array.create(values);
+    }
+
+    public Document toDocument(TableChange tableChange) {
+        Document document = Document.create();
+
+        document.setString("type", tableChange.getType().name());
+        document.setString("id", tableChange.getId().toDoubleQuotedString());
+        document.setDocument("table", toDocument(tableChange.getTable()));
+        return document;
+    }
+
+    private Document toDocument(Table table) {
+        Document document = Document.create();
+
+        document.set("defaultCharsetName", table.defaultCharsetName());
+        document.set("primaryKeyColumnNames", Array.create(table.primaryKeyColumnNames()));
+
+        List<Document> columns =
+                table.columns().stream().map(this::toDocument).collect(Collectors.toList());
+
+        document.setArray("columns", Array.create(columns));
+
+        return document;
+    }
+
+    private Document toDocument(Column column) {
+        Document document = Document.create();
+
+        document.setString("name", column.name());
+        document.setNumber("jdbcType", column.jdbcType());
+
+        if (column.nativeType() != Column.UNSET_INT_VALUE) {
+            document.setNumber("nativeType", column.nativeType());
+        }
+
+        document.setString("typeName", column.typeName());
+        document.setString("typeExpression", column.typeExpression());
+        document.setString("charsetName", column.charsetName());
+
+        if (column.length() != Column.UNSET_INT_VALUE) {
+            document.setNumber("length", column.length());
+        }
+
+        column.scale().ifPresent(s -> document.setNumber("scale", s));
+
+        document.setNumber("position", column.position());
+        document.setBoolean("optional", column.isOptional());
+        document.setBoolean("autoIncremented", column.isAutoIncremented());
+        document.setBoolean("generated", column.isGenerated());
+
+        // BEGIN FLINK MODIFICATION
+        document.setArray("enumValues", column.enumValues().toArray());
+        // END FLINK MODIFICATION
+
+        return document;
+    }
+
+    @Override
+    public TableChanges deserialize(Array array, boolean useCatalogBeforeSchema) {
+        TableChanges tableChanges = new TableChanges();
+
+        for (Entry entry : array) {
+            TableChange change =
+                    fromDocument(entry.getValue().asDocument(), useCatalogBeforeSchema);
+
+            if (change.getType() == TableChangeType.CREATE) {
+                tableChanges.create(change.getTable());
+            } else if (change.getType() == TableChangeType.ALTER) {
+                tableChanges.alter(change.getTable());
+            } else if (change.getType() == TableChangeType.DROP) {
+                tableChanges.drop(change.getTable());
+            }
+        }
+
+        return tableChanges;
+    }
+
+    private static Table fromDocument(TableId id, Document document) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method calls 26 other methods. By comparison, 98% of the methods in the CodeGuru reference dataset use fewer. This indicates the method is highly coupled with other methods. A method that is highly coupled with other methods is difficult to understand and its behavior might change unexpectedly when one of its referenced methods is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this method or break it into multiple methods.